src/components/ArticleDetail.jsx

```diff
@@ -9,11 +9,33 @@ export default function ArticleDetail (){
     const [article, setArticle] = useState({})
     const {articleId} = useParams()
     const [isLoading, setIsLoading] = useState(true)
+    const [votes, setVotes] = useState(0)
+
+    const handleDownVote = () => {
+        setVotes((currVotes) => currVotes - 1)
+        patchArticle(articleId, {inc_votes: -1}).then((article) => {
+            setVotes(article.votes)
```

Comment:
Instead of updating the votes twice in the handleUpVote and handleDownVote functions, we can optimize the process to reduce glitchiness and improve efficiency. Currently I am getting some glitchy behaviour, but this should hopefully resolve the matter...

Here's a revised approach:
```
const handleUpVote = () => {
    setVotes((currVotes) => currVotes + 1);
    patchArticle(articleId, { inc_votes: 1 })
        .catch((err) => {
            setVotes((currVotes) => currVotes - 1); // Revert to previous votes if patch fails
        });
};

const handleDownVote = () => {
    setVotes((currVotes) => currVotes - 1);
    patchArticle(articleId, { inc_votes: -1 })
        .catch((err) => {
            setVotes((currVotes) => currVotes + 1); // Revert to previous votes if patch fails
        });
};
```
This approach utilizes optimistic rendering by immediately updating the votes upon user interaction. If the patch operation fails, the votes are reverted to their previous state to maintain consistency and prevent glitchiness.